sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/GeneralScalarExpression.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.expressions;
+
+import java.io.Serializable;
+
+import org.apache.spark.annotation.Evolving;
+
+/**
+ * The general representation of SQL scalar expressions, which contains the upper-cased
+ * expression name and all the children expressions.
+ * <p>
+ * The currently supported expressions:
+ * <table border="1">
+ *  <tr>
+ *   <th>Expression name</th>
+ *   <th>SQL scalar expression</th>
+ *   <th>Since version</th>
+ *  </tr>
+ *  <tr>
+ *   <td>IS_NULL</td>
+ *   <td><pre>expr IS NULL</pre></td>
+ *   <td>3.3.0</td>
+ *  </tr>
+ *  <tr>
+ *   <td>IS_NOT_NULL</td>
+ *   <td><pre>expr IS NOT NULL</pre></td>
+ *   <td>3.3.0</td>
+ *  </tr>
+ *  <tr>
+ *   <td>=</td>
+ *   <td><pre>expr1 = expr2</pre></td>
+ *   <td>3.3.0</td>
+ *  </tr>
+ *  <tr>
+ *   <td>!=</td>
+ *   <td><pre>expr1 != expr2</pre></td>
+ *   <td>3.3.0</td>
+ *  </tr>
+ *  <tr>
+ *   <td><=></td>
+ *   <td><pre>expr1 <=> expr2</pre></td>
+ *   <td>3.3.0</td>
+ *  </tr>
+ *  <tr>
+ *   <td><</td>
+ *   <td><pre>expr1 < expr2</pre></td>
+ *   <td>3.3.0</td>
+ *  </tr>
+ *  <tr>
+ *   <td><=</td>
+ *   <td><pre>expr1 <= expr2</pre></td>
+ *   <td>3.3.0</td>
+ *  </tr>
+ *  <tr>
+ *   <td>></td>
+ *   <td><pre>expr1 > expr2</pre></td>
+ *   <td>3.3.0</td>
+ *  </tr>
+ *  <tr>
+ *   <td>>=</td>
+ *   <td><pre>expr1 >= expr2</pre></td>
+ *   <td>3.3.0</td>
+ *  </tr>
+ *  <tr>
+ *   <td>+</td>
+ *   <td><pre>expr1 + expr2</pre></td>
```

Comment:
Now, this comes to the tricky part. If ANSI mode is off, `expr1 + expr2` has a different behavior in Spark compared to other databases, because it won't fail for overflow.

One simple fix is: We don't translate math operations to v2 expression if ANSI mode is off. Or we can have a new expression name for ANSI-off add.